src/lib/route_rule.rs

```diff
@@ -0,0 +1,206 @@
+use crate::netlink::parse_as_ipv4;
+use crate::netlink::parse_as_ipv6;
+use crate::route::AddressFamily;
+use crate::route::RouteProtocol;
+use crate::NisporError;
+use futures::stream::TryStreamExt;
+use netlink_packet_route::rtnl::rule::nlas::Nla;
+use netlink_packet_route::RuleMessage;
+use rtnetlink::new_connection;
+use rtnetlink::IpVersion;
+use serde_derive::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+const FR_ACT_TO_TBL: u8 = 1;
+const FR_ACT_GOTO: u8 = 2;
+const FR_ACT_NOP: u8 = 4;
+const FR_ACT_RES3: u8 = 8;
+const FR_ACT_RES4: u8 = 16;
+const FR_ACT_BLACKHOLE: u8 = 32;
+const FR_ACT_UNREACHABLE: u8 = 64;
+const FR_ACT_PROHIBIT: u8 = 128;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum RuleAction {
+    /* Pass to fixed table */
+    Table,
+    /* Jump to another rule */
+    Goto,
+    /* No operation */
+    Nop,
+    Res3,
+    Res4,
```

Comment:
Unless you know what is `Res3` and `Res4` is, please remove it.

There is not document in kernel mentioned this. And iproute2 has not such code neither.

Reply:
Done, removed.